### PR TITLE
Module 3

### PR DESCRIPTION
**Pull request by ncaunhye**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/tests/test_view_model.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import os 
+from todo_app.todo_item import TodoItem
+
+
+@pytest.fixture
+def view_model():
+    _DEFAULT_CARDS = [
+        {
+            "idShort": 1,
+            "idList": os.getenv("to_do_list"),
```
Comment (BenRamchandani) [inline todo_app/tests/test_view_model.py:13]: These tests depend on these environment variables being set, the list ids have to be different or the logic in todo_item.py will not work. If you run e.g. `poetry run pytest todo_app/tests/test_view_model.py` they won't be.

You could load `.env.test` like you do for the integration tests, or manually set them at the start of this function.

File: todo_app/view_model.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return self._items
+
+    @property
+    def todo_items(self):
+        todo_items = []
+        for item in self._items:
+            if item.status == "To Do":
+                todo_items.append(item)
+            else:
+                continue
```
Comment (BenRamchandani) [inline todo_app/view_model.py:19]: These two lines don't do anything

File: .env.template [hunk 3]
```diff
@@ -3,4 +3,6 @@ FLASK_APP=todo_app/app
 FLASK_ENV=development
 
 # Change the following values for local development.
-SECRET_KEY=secret-key
+test_key=""
+
```
Comment (BenRamchandani) [inline .env.template:7]: Can you add placeholders for the other variables in here too, like in .env.test?

File: todo_app/tests/test_view_model.py (hunk 1 above)
Comment (ncaunhye) [reply todo_app/tests/test_view_model.py]: I have now loaded .env.test in my test_view_model. Please could you review. Thanks

File: todo_app/view_model.py (hunk 2 above)
Comment (ncaunhye) [reply todo_app/view_model.py]: Ive removed all the else, continues. Thanks

**Review by BenRamchandani — APPROVED** [review]
Looks great, thanks for doing all the markups!

I've left some more (completely optional) comments to look at if you'd like. Whether you do them or not go ahead and merge when you're ready 🙂

File: README.md [hunk 4]
```diff
@@ -50,3 +50,21 @@ You should see output similar to the following:
...
+The .env.template file shows the values that you will need to extract from the Trello board/card data and this can be found using the Trello API - "https://api.trello.com/"
+
+
+## Pytest 
+
+Before running the app you can run unit and integration tests to see if the app works, the tests are found in the tests folder. 
+
+To run unit tests run the following command - 
+```
+$ pytest todo_app/tests
```
[the remark this answers is not in the document]
Comment (ncaunhye) [reply README.md:69]: done

File: todo_app/data/session_items.py [hunk 5]
```diff
@@ -62,8 +65,53 @@ def save_item(item):
...
+                id = card["id"]
+                url = f"https://api.trello.com/1/cards/{id}"
+                data = {"idList": os.getenv("to_do_list")}
+                response = requests.put(url, data=data, params=query)
+        if string_select_update == "Complete":
+            if card["idShort"] == int(item_id):
+                id = card["id"]
+                url = f"https://api.trello.com/1/cards/{id}"
+                data = {"idList": os.getenv("complete_list")}
+                response = requests.put(url, data=data, params=query)
```
[the remark this answers is not in the document]
Comment (ncaunhye) [reply todo_app/data/session_items.py:107]: will look into this in the future

**Pull request merged**